tests/conftest.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+import json
+import mock
+
+from scc_hypervisor_collector.api import ConfigManager, CollectionScheduler, HypervisorCollector
+from scc_hypervisor_collector import cli
+
+
+@pytest.fixture
+def config_manager(request):
+    marker = request.node.get_closest_marker("config")
+    return ConfigManager(config_file=marker.args[0], config_dir=marker.args[1])
+
+@pytest.fixture
+def backendid(request):
+    if getattr(request, 'param', None):
+        backendid = request.param
+    return backendid
```

Comment:
Maybe it is a fixture thing, but this return statement appears to be returning a variable that is only defined/initialised in the then case of the if getattr(...) check... What happens if request doesn't have a param attr? Or will it always have a param attr, in which case you can just return request.param without the if check altogether?

Reply:
This depends on the test case. The existence of this parameter request.param depends on whether the test defines a parameter "backendid".

Reply:
But you always return backendid, which normally would result in a python exception if you tried to return a variable that never got defined if `getattr(request, 'param', None)` returns a `None`, or a value of 0, or an empty string, or anything equivalent to False...
Note that a Python function implicitly returns `None` so you could just change the body of the if getattr(...) to return request.param, and that would be valid and unambiguous Python ;-)